libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2816,19 +2855,20 @@ void IRGeneratorForStatements::appendBareCall(
 
 	templ("address", IRVariable(_functionCall.expression()).part("address").name());
 
+	std::string const callPrefix = eof ? "ext" : "";
 	if (funKind == FunctionType::Kind::BareCall)
 	{
 		templ("value", funType.valueSet() ? IRVariable(_functionCall.expression()).part("value").name() : "0");
-		templ("call", "call");
+		templ("call", callPrefix + "call");
 	}
 	else
 	{
 		solAssert(!funType.valueSet(), "Value set for delegatecall or staticcall.");
 		templ("value", "");
 		if (funKind == FunctionType::Kind::BareStaticCall)
-			templ("call", "staticcall");
+			templ("call", callPrefix + "staticcall");
 		else
-			templ("call", "delegatecall");
+			templ("call", callPrefix + "delegatecall");
 	}
 
 	if (funType.gasSet())
```

Comment:
Please assert that in EOF `gas` is not set. Same in `appendExternalFunctionCall()`.

BTW, for this assert to be valid, we must also disallow setting `gas` on external calls via analysis. Please remember to do that later. Unless the plan is to just ignore `gas` on EOF for easier transition from legacy? But in that case we should still at least issue a warning in analysis.

Reply:
In this PR?

Reply:
I think that the asserts should be added already here as a quick sanity check. The analysis error or warning can be done separately.

Reply:
Asserts added. I meant the warning.